sdk/core/azure-core/src/http/user_agent.cpp

```diff
@@ -164,7 +165,8 @@ namespace Azure { namespace Core { namespace Http { namespace _detail {
     }
 
     static std::string const osVer = GetOSVersion();
-    telemetryId << "azsdk-cpp-" << componentName << "/" << componentVersion << " (" << osVer << ")";
+    telemetryId << "azsdk-cpp-" << componentName << "/" << componentVersion << " (" << osVer << " "
+                << "Cpp" << cplusplusValue << ")";
```

Comment:
Just a question, it's probably fine as is.
Does it make sense to put cpp version in the same parenthesis next to the os version? Maybe put separately outside the parenthesis?

Reply:
Also, might be worth creating a more unique prefix, for ease of finding the string literal.
Maybe `Cpp_Version_`

Reply:
> Just a question, it's probably fine as is. Does it make sense to put cpp version in the same parenthesis next to the os version? Maybe put separately outside the parenthesis?

Chiming in as the C++ architect: [The guidelines](https://azure.github.io/azure-sdk/general_azurecore.html#user-agent-format) do not explicitly specify that the platform specific elements (OS, OS version, etc) are contained within parenthesis, but all the examples are inside the parenthesis (python includes the python runtime version formatted as a package version using the string `Python/3.73`).

One nit that I noticed reading the guidelines: The guidelines actually specify you need to include the package AND its dependent packages (including their versions) which would require a fairly significant change to how we handle the telemetry policy.

Reply:
We should clarify that statement in the guidelines. I want to be consistent with other SDK packages. That might mean storage includes storage, storage common, and core for example. It could also mean storage and core. 
We should NOT be listing the additional dependencies (CURL, WinHttp are a reasonable exception to that rule).

@antkmsft Can you research other language SDKs. Lets aim to be consistent in the approach.

Reply:
The current C++ Telemetry guidelines are [here](https://azure.github.io/azure-sdk/cpp_implementation.html#telemetry).

.Net, Java, Typescript, Rust, and Go guidelines are mute on user agent. Some of them have "TODO" sections indicating that there is an intention to define guidelines.
Python's guidelines are [here](https://azure.github.io/azure-sdk/python_implementation.html#telemetry).

Reply:
We include the value in parenthesis, together with OS information, because that is where all the extra information is going. We are not including the version of Azure SDK package dependencies. We could, if we needed to, but we don't.
I prefer `Cpp` to `Cpp_Version_`, because it is shorter but it is essentially the same.